File: Java/프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int result = 0;
        for (int i = 1; i<=number; i++) {
            int target = 0;
            for (int j = 1; j*j<=i; j++) {
                if (j * j == i) {
                    target++;
                } else if (i % j == 0) {
                    target += 2;
                }
            }
            if (target > limit) {
                target = power;
            }
            result += target;
        }
        
        return result;
    }
}

File: Swift/프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.swift
import Foundation

func solution(_ number:Int, _ limit:Int, _ power:Int) -> Int {
    var result = 0
    for i in stride(from:0, through: number, by:1) {
        var target = 0
        var j = 1
        while j*j <= i {
            if (j * j) == i {
                target += 1
            } else if (i % j == 0) {
                target += 2
            }
            j += 1
        }
        
        if target > limit {
            target = power
        }
        result += target
    }
    return result
}